### PR TITLE
CAMEL-11834: Add tests for activemq/aggregate/any23/artemis/csimple/cxf

### PR DESCRIPTION
**Pull request by essobedo**
## Motivation

Most of the Camel examples don't have any test such that they need to be launched manually to ensure that they still work which is not convenient.

## Modifications:

* Refactor the code of `aggregate-dist` to be able to call it from a test for the sake of simplicity
* Fix the max heap size for the test of `artemis-large-messages` to ensure that it will fail in case the spooling doesn't work as expected
* Use ActiveMQ / ActiveMQ Artemis in embedded mode in the tests for the sake of simplicity
* Use Arquillian for integration tests that need Tomcat (activemq-tomcat and cxf-tomcat)
* Fix the readme.doc (wording, grammar, out dated info, typos)
* Fix deprecated code
* Remove the file `src/main/data/foo.properties` from csimple examples as it is not used
* Modify the route used for `csimple-joor` to have expressions that are more jOOR specific
* Remove useless namespace definitions
* Refresh the code for any23 as it did not return anything
* Improve existing code by using `try-with-resources` statement to properly close the resources
* Remove duplicate dependencies in pom files
* Add new folders created by the tests to be ignored by git 
* Upgrade the version of Arquillian to be able to write test in JUnit 5
* Upgrade the version of Tomcat 9

### CODE REVIEW

File: examples/aggregate-dist/src/main/java/org/apache/camel/example/Application.java [hunk 1]
```diff
@@ -34,55 +34,71 @@
...
+import java.sql.Statement;
 import java.util.Arrays;
 import java.util.Queue;
 import java.util.UUID;
-import java.util.concurrent.ConcurrentLinkedQueue;
-import java.util.concurrent.CountDownLatch;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.TimeUnit;
+import java.util.concurrent.*;
```
Comment (oscerd) [inline examples/aggregate-dist/src/main/java/org/apache/camel/example/Application.java:41]: Please don't use the * import. Just fine tuned imports.

File: examples/activemq-tomcat/src/test/java/org/apache/camel/example/ActiveMQTomcatIT.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import org.apache.activemq.ActiveMQConnectionFactory;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit5.ArquillianExtension;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.shrinkwrap.resolver.api.maven.Maven;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import javax.jms.*;
```
Comment (oscerd) [inline examples/activemq-tomcat/src/test/java/org/apache/camel/example/ActiveMQTomcatIT.java:28]: Please don't use the * import. Just fine tuned imports.

**Review by oscerd — CHANGES_REQUESTED** [review]
Two minor codestyle findings

**Comment by essobedo** [comment]
@oscerd fixed, thx for the feedback. 🙏 
Is-it what was expected?

**Comment by oscerd** [comment]
Something like this, but many of the examples are dependant on external system, so it won't be really easy to write tests for them.

**Comment by essobedo** [comment]
> Something like this, but many of the examples are dependant on external system, so it won't be really easy to write tests for them.

@oscerd Is it acceptable to use testcontainer for those tests?

**Comment by oscerd** [comment]
Yes, we use testcontainers extensively in the Camel test infra modules

**Comment by essobedo** [comment]
For this task, do you prefer several PRs of this size or one huge PR?

**Comment by oscerd** [comment]
Whatever works better for you and your time. Thanks.

**Review by davsclaus — APPROVED** [review]

**Comment by essobedo** [comment]
@oscerd If so, I will do several PRs if you don't mind

**Review by oscerd — APPROVED** [review]

**Comment by essobedo** [comment]
@oscerd Hi 👋 , I'm not yet familiar with the validation process. How many approvals are needed to merge the PR?

**Comment by oscerd** [comment]
I just need to find the time to test e merge

**Comment by essobedo** [comment]
BTW you will notice that most of the time, I don't really test the output, I rather test that the messages are processed with success. I assumed that it could be enough in this case but let me know if you want me to test the output.

**Pull request merged**